### ISSUE TITLE
Is TSDX still maintained?

### ISSUE THREAD

**Issue by y-nk** [formium/tsdx#1065]
It's not an offensive question, but a more a friendly business-related inquiry.

I'm really loving the project and the overall initiative, but on my side  I can't afford to base my work on it if it's dying/going to die anytime soon. I'm not asking for an answer with some commitment of any sort, but just a general pulse and the overall mindset of the authors. I understand maintaining an open-source project on spare time is a huge work and I've been reading numerous stories about it, so please do consider I'm really understanding if you guys are busy/having a hard time/etc. I'm currently asserting basing 3 majors libraries of the company I work in, which will be under ~15~20 frontend guys of various background and so I need to confidence that this tool will be solid enough to support them for the years to come.

Thanks for the work accomplished so far anyway :)

**Comment by adamfratino** [comment]
#1058

**Comment by y-nk** [comment]
Thanks @adamfratino ! Honestly if it's not being maintained it would nice for all people who jumped on board to actually give it back to the community. I understand @jaredpalmer has intent to earn money with turborepo but it's an ass move to lock us up like this (leaving this OSS project dead in favor of a paid one). Of course, we could be doing forks, but there would be "the battle of the forks" and it'd be rather nice to pass ownership to someone (not me) who has intent to drive this project forward for all the people already in.

**Comment by adamfratino** [comment]
@y-nk +100!

My team and I chose TSDX for an internal project last December. We're currently being forced to have conversations about replatforming because we don't have faith in the future of TSDX.

The only issue I have with all this is the lack of transparency. I've also lost faith in recommending/using any future tools from Jared based on this experience. It's simply too risky.

**Comment by luigiminardim** [comment]
Any alternatives?

**Comment by aladdin-add** [comment]
as alternatives: 
* a fork of tsdx: https://www.npmjs.com/package/@weiran.zsd/tsdx
* microbundle
* just use rollup/eslint/prettier/typescript/jest/...

**Comment by Hocdoc** [comment]
I find this TS starter template useful: https://www.metachris.com/2021/04/starting-a-typescript-project-in-2021/

**Comment by AntonyFagundez** [comment]
I would like to contribute. If its possible...

**Comment by MarcelCutts** [comment]
@y-nk Do let us know if manage to clinch that multi-year SLA for an open source project.  While not a traditional approach, given the impressive size of your engineering pool I am sure corporate will be happy paying to retain key respository contributors.

**Comment by y-nk** [comment]
@MarcelCutts Is that sarcasm?

As said I'm grateful for the work accomplished and I won't change my position on that, but that the same time you gotta understand that when you put something out there and it's _that_ great, people using it are gonna have a minimum of expectations ; and _it's okay_ to not want to carry that burden, no one wants to ever bully you into forced labor... but the least people want is simple/minimum communication.

"is the project dead in favor of a paid one?" 

if so, simply put a message in the readme or archive this one would suffice to be clear about it so that newcomers won't think "oh great! i can start my codebase with it." ; again i'm not judging, if that's the destiny of this repo, then be it.

...and if you ever care on the TypeScript community as much as you did when you started this project... as proposed, maybe ask someone "who has time and dedication to continue what i started because i just don't have the capacity at the moment" and add them as contributor so that the project can outlive you and continue to help other people.

**Comment by SayakMukhopadhyay** [comment]
A big problem is the typescript official website pointing to this tool as a starter https://www.typescriptlang.org/docs/bootstrap. I will try and raise an issue with them to remove and maybe suggest an alternative.

**Comment by JustFly1984** [comment]
In my @react-google-maps/api library we had replaced tsdx with rollup this month and it is doing great. If somebody need an example, look into my package source code.

**Comment by SayakMukhopadhyay** [comment]
For the past couple of weeks, I have started working on a new library which aims to bootstrap a typescript project quickly with some sane defaults. For now, I am only focusing on bootstrapping and not other things. See https://github.com/SayakMukhopadhyay/ts-boot.

The reason I am starting a new project is to eliminate tech debt from tsdx author. Contributions and inputs welcome.

**Comment by iamandrewluca** [comment]
https://github.com/unjs/siroc

**Comment by bke-daniel** [comment]
Sad to see this situation, but we really appreciated `tsdx` at @we-io & @hotelery.
Thanks for to good work @jaredpalmer! 

Time to move on for us. Thank you!

**Comment by iamandrewluca** [comment]
I think this one is also good https://github.com/unjs/siroc

**Comment by iamandrewluca** [comment]
Also nowadays one can use https://nx.dev/ it works for single or monorepo

**Comment by colinrobertbrooks** [comment]
Today it was announced that [Turborepo is joining Vercel](https://turborepo.org/blog/joining-vercel) and [Turborepo CLI is now open-source](https://github.com/vercel/turborepo).

**Comment by SayakMukhopadhyay** [comment]
So, can turborepo be considered as a spiritual successor to TSDX?

**Comment by philals** [comment]
Does anyone have a good way to eject?

**Comment by y-nk** [comment]
@colinrobertbrooks is that a non-explicit way to confirm that tsdx should be considered as deprecated/not-maintained anymore?

**Comment by colinrobertbrooks** [comment]
@y-nk, no. It's just information as Turborepo has been discussed above.

**Comment by MikeRippon** [comment]
> as alternatives:
> 
>     * a fork of tsdx: https://www.npmjs.com/package/dts-cli
>     * microbundle
>     * just use rollup/eslint/prettier/typescript/jest/...

+1 for microbundle. Took me days to get up and running with tsdx because of all the outdated dependencies. Dropped microbundle in and it worked straight away.

**Comment by 0xdevalias** [comment]
Another alternative suggestion, found at: https://github.com/hoeck/simple-runtypes/issues/63

> https://github.com/egoist/tsup looks like a good candidate

- https://tsup.egoist.sh/
- https://github.com/egoist/tsup

**Comment by awxalbert** [comment]
I'm wondering if Jared is gonna give up TSDX, wouldn't it be nice to add some deprecation warning in README and point new comers to an alternative like [tsup](https://github.com/egoist/tsup) ? I don't think just leave the repo here a good idea.

**Comment by AntonyFagundez** [comment]
> +1 for a note explaining this isn't being maintained (if that is the case). Spent a frustrating day-and-a-half trying to get it to play with React 18 and github actions; now finding parcel won't hot reload which is really the whole point!

In my job we need to move to plain rollup, typescript and everything. Because We initialize with this our Design System Implementation in React, and we need to keep update. This library has not been updated for more than a year. 

I recommend the same alternatives that @aladdin-add.

> as alternatives:
> * a fork of tsdx: https://www.npmjs.com/package/dts-cli
> * microbundle
> * just use rollup/eslint/prettier/typescript/jest/...

But specially the last one because the full control of styles. (CSS/SCSS Modules) and injection with [rollup-plugin-styles](https://www.npmjs.com/package/rollup-plugin-styles) if you want to make a component library.

**Comment by sandtreader** [comment]
In case it helps anyone, I went a different way - reverted to trusty old create-react-app, refactored into lib/ and example/ and added back microbundle to publish it.  The empty shell is available here if anyone wants to look at it:

https://github.com/sandtreader/rafiki/releases/tag/EMPTY-SHELL

The resulting (empty, useless, for now!) npm module is here:

https://www.npmjs.com/package/@sandtreader/rafiki

I can imagine at some point ejecting CRA and going more custom but this will be fine for now.

**Comment by moishinetzer** [comment]
I tried to use this but couldn't get it to work in the context of a component library for react.

So I created my own solution, and I thought I would drop it here to see if it could help anyone:

- https://github.com/moishinetzer/pbandj

**Comment by y-nk** [comment]
I should have done this long ago (not sure why i didn't), but seeing new comments still submitted here to this day, it started to feel necessary.

There's now a PR to propose a simple message about this being a zombie repo. There's little to do now to make a change. A single review and a merge, and we'll close this issue for good.

I'm not sure who/how to push for this. I hope maintainers will receive a notification, care to read and won't take it the wrong way. We're just trying to help here.

**Comment by nyngwang** [comment]
This is open-source: leave-any-time-of-course.

**Comment by gsouf** [comment]
For who is wondering - these days a good way to start a project is to use a vite provided template, they have a bunch of those:
https://github.com/vitejs/vite/tree/main/packages/create-vite

example:

```
npm create vite@latest my-app -- --template react-ts
```
